### PR TITLE
Move Deposits.upsert into its own service

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary

We need this to avoid a circular dependency where events needs to
reference the Deposits service.

## Testing Plan

Run tests

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: src/events/deposits.service.ts [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
-import { UpsertDepositsOperationDto } from './dto/upsert-deposit.dto';
-import { EventsService } from './events.service';
 
 @Injectable()
 export class DepositsService {
   constructor(
     private readonly config: ApiConfigService,
     private readonly prisma: PrismaService,
-    private readonly events: EventsService,
     private readonly users: UsersService,
```
Comment (rohanjadvani) [inline src/events/deposits.service.ts:15]: Can this be removed?

**Review by brekk — APPROVED** [review]

File: src/events/deposits.service.ts (hunk 1 above)
Comment (NullSoldier) [reply src/events/deposits.service.ts]: Removed that and config

File: src/events/deposits.service.ts [hunk 2]
```diff
@@ -1,25 +1,15 @@
...
-  constructor(
-    private readonly config: ApiConfigService,
-    private readonly prisma: PrismaService,
-    private readonly events: EventsService,
-    private readonly users: UsersService,
-  ) {}
+  constructor(private readonly prisma: PrismaService) {}
 
   async find(id: number): Promise<Deposit | null> {
     return await this.prisma.deposit.findUnique({
```
Comment (rohanjadvani) [inline src/events/deposits.service.ts:15]:
```suggestion
    return this.prisma.deposit.findUnique({
```

**Review by rohanjadvani — APPROVED** [review]

File: src/events/deposits.service.ts (hunk 2 above)
Comment (NullSoldier) [reply src/events/deposits.service.ts]: done

**Pull request merged**